backend/src/apiserver/server/api_converter.go

```diff
@@ -316,113 +309,38 @@ func toApiPipelines(pipelines []*model.Pipeline) []*apiv2beta1.Pipeline {
-		if apiResourceRefs == nil {
-			apiResourceRefs = make([]*apiv1beta1.ResourceReference, 0)
-		}
-		newPipelineId := apiPipelineV1.GetId()
-		newPipelineIdRef := getPipelineIdFromResourceReferencesV1(apiResourceRefs)
-		if newPipelineId == "" && newPipelineIdRef != "" {
-			newPipelineId = newPipelineIdRef
-		} else if newPipelineId != newPipelineIdRef {
-			if newPipelineIdRef != "" {
-				apiResourceRefsNew := make([]*apiv1beta1.ResourceReference, 0)
-				for _, resRef := range apiResourceRefs {
-					if resRef.GetKey() != nil {
-						if resRef.GetKey().GetId() != newPipelineIdRef {
-							apiResourceRefsNew = append(apiResourceRefsNew, resRef)
-						}
-					}
-				}
-				apiResourceRefs = apiResourceRefsNew
-			}
-		}
-		apiPipelineVersionV1.ResourceReferences = append(
-			apiResourceRefs,
-			&apiv1beta1.ResourceReference{
-				Key: &apiv1beta1.ResourceKey{
-					Id:   newPipelineId,
-					Type: apiv1beta1.ResourceType_PIPELINE,
-				},
-				Relationship: apiv1beta1.Relationship_OWNER,
-			},
-		)
-
-		// Infer Parameters from the parent if they are missing or invalid
-		if apiPipelineVersionV1.GetParameters() == nil && apiPipelineV1.GetParameters() != nil {
-			apiPipelineVersionV1.Parameters = apiPipelineV1.GetParameters()
-		}
-
-		// Infer CodeSourceUrl from the parent if it is missing or invalid
-		tempUrl := ""
-		if apiPipelineVersionV1.GetCodeSourceUrl() == "" {
-			if apiPipelineVersionV1.GetPackageUrl() != nil {
-				tempUrl = apiPipelineVersionV1.GetPackageUrl().GetPipelineUrl()
-			}
-		}
-		if tempUrl == "" {
-			if apiPipelineV1.GetUrl() != nil {
-				apiPipelineVersionV1.CodeSourceUrl = apiPipelineV1.GetUrl().GetPipelineUrl()
-			}
-		}
-
-		return toModelPipelineVersion(apiPipelineVersionV1)
 	case *apiv2beta1.PipelineVersion:
 		apiPipelineVersionV2 := p
-		spec, err := pipelineSpecStructToYamlString(apiPipelineVersionV2.GetPipelineSpec())
-		if err != nil {
-			return nil, util.NewInternalServerError(err, "Failed to convert API pipeline version to internal pipeline version representation due to pipeline spec conversion error")
+		if apiPipelineVersionV2.GetPackageUrl() == nil || len(apiPipelineVersionV2.GetPackageUrl().GetPipelineUrl()) == 0 {
+			return nil, util.NewInvalidInputError("Failed to convert v2beta1 API pipeline version to its internal representation due to missing pipeline URL")
 		}
-		pipelineSpec = spec
-		uuid = apiPipelineVersionV2.GetPipelineVersionId()
-		createTime = apiPipelineVersionV2.GetCreatedAt().GetSeconds()
 		name = apiPipelineVersionV2.GetDisplayName()
-		parameters = ""
 		pipelineId = apiPipelineVersionV2.GetPipelineId()
-		codeUrl = apiPipelineVersionV2.GetPackageUrl().GetPipelineUrl()
+		pipelineUrl = apiPipelineVersionV2.GetPackageUrl().GetPipelineUrl()
+		codeUrl = apiPipelineVersionV2.GetCodeSourceUrl()
 		description = apiPipelineVersionV2.GetDescription()
 	default:
 		return nil, util.NewUnknownApiVersionError("PipelineVersion", p)
 	}
 	return &model.PipelineVersion{
```

Comment:
Nit: Is there a specific reason we are not using `toModelPipelineVersion()` here?

Reply:
This code is inside `toModelPipelineVersion`, it can't recursively calling itself without a stop condition.

Reply:
Sorry, I misread. XD I thought this was inside `createPipelineVersion()`.